action.yml

```diff
@@ -25,22 +25,33 @@ inputs:
     description: 'ID of the distributionManagement repository in the pom.xml
        file. Default is `github`'
     required: false
+    default: 'github'
   server-username:
     description: 'Environment variable name for the username for authentication
        to the Apache Maven repository. Default is $GITHUB_ACTOR'
     required: false
+    default: 'GITHUB_ACTOR'
```

Comment:
This should be `${{ GITHUB_ACTOR }}`

Env variables need to be wrapped in the special syntax so they correctly get resolved. Example: https://github.com/actions/setup-python/blob/7a69c2bc7dc38832443a11bc7c2550ba96c6f45c/action.yml#L13

Reply:
Are you sure? The defaults look like they were originally the string literals:

https://github.com/actions/setup-java/blob/d920b7da5f18f6c632513f44c068598875961bbb/src/auth.ts#L11-L12

That way they would be written to the settings.xml file with environment variables names rather than the actual environment variable values:

https://github.com/actions/setup-java/blob/d920b7da5f18f6c632513f44c068598875961bbb/src/auth.ts#L54-L55

From the README:

https://github.com/actions/setup-java/blob/master/README.md#publishing-using-apache-maven

```xml
<servers>
    <server>
      <id>github</id>
      <username>${env.GITHUB_ACTOR}</username>
      <password>${env.GITHUB_TOKEN}</password>
    </server>
</servers>
```

I believe if I put the wrapping there, the `settings.xml` file would look like:

```xml
<servers>
    <server>
      <id>github</id>
      <username>jaredpetersen</username>
      <password>mysupersecrettokeninplaintext</password>
    </server>
</servers>
```

Reply:
Actually, I think it would look like:

```xml
<servers>
    <server>
      <id>github</id>
      <username>${env.jaredpetersen}</username>
      <password>${env.mysupersecrettokeninplaintext}</password>
    </server>
</servers>
```

Reply:
From my test output, it looks like it's correct as-is. See the `Output settings.xml` step [here](https://github.com/jaredpetersen/setup-java-gpg/runs/876184800?check_suite_focus=true).

Reply:
You're right! I forgot that you can use interpolation and that env variables will be later evaluated.
Something like `${env.HOME}` can be directly in the settings file: https://maven.apache.org/settings.html#quick-overview